baseplate/experiments/providers/__init__.py

```diff
@@ -0,0 +1,137 @@
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import logging
+import time
+
+from .forced_variant import ForcedVariantExperiment
+from .legacy import LegacyExperiment
+from ...features.feature import feature_flag_from_config
+
+logger = logging.getLogger(__name__)
+
+
+class ExperimentManager(object):
+
+    def __init__(self, id, name, owner, experiment, enabled=True,
+                 feature_flag=None):
+        self.id = id,
+        self.name = name
+        self.owner = owner
+        self._experiment = experiment
+        self._enabled = enabled
+        self._feature_flag = feature_flag
+
+    def enabled(self, **args):
+        if not self._enabled:
+            return False
+
+        if self._feature_flag is None:
+            return True
+
+        if self._feature_flag.enabled(**args):
+            return True
+
+        return False
+
+    def variant(self, **args):
+        return self._experiment.variant(**args)
+
+    def should_log_bucketing(self):
+        return self._experiment.should_log_bucketing()
+
+    def event_params(self):
+        event_params = {
+            "experiment_id": self.id,
+            "experiment_name": self.name,
+            "experiment_owner": self.owner,
+        }
+        return event_params
+
+
+def experiment_from_config(config):
+    experiment_type = config["type"].lower()
+    experiment_id = config["id"]
+    name = config["name"]
+    owner = config.get("owner")
+    experiment_config = config["experiment"]
+    expiration = config["expires"]
+    if int(time.time()) > expiration:
+        logger.warning(
```

Comment:
I've been thinking about these more and I think I'm going to cut out these warnings and the expiration enforcement from baseplate, I think we can build a better system to bug people about these things within the experiment service that we use to manage the configs.